### ISSUE TITLE
Directory.CreateDirectory($@"C: \Users\{ username}\") creates a folder name with pure whitespaces

### ISSUE THREAD

**Issue by lifengl** [dotnet/runtime#38584]
It is reported by a customer when there is a typo in the code. Running the code would create a folder inside the current running dictionary in C: with pure whitespace characters, which doesn't work with Windows explorer and other tools.

This is different from the behavior of the code in NET Framework. Although this might be a valid name in NTFS, not sure whether it is an expected behavior here.

**Comment by danmosemsft** [comment]
Specifically 
```c#
Directory.SetCurrentDirectory(@"c:\foo");
Directory.CreateDirectory($@"c: \oops")
```
 (with space after `C:`) creates a folder named `c:\foo\ ` (single space)

**Comment by Clockwork-Muse** [comment]
Side note, why are they manually creating what appear to be user directories?

**Comment by danmosemsft** [comment]
Good question, it would be easy to get the ACL's wrong (besides the question of why)

**Comment by Clockwork-Muse** [comment]
Maybe they really wanted the directory to really match the username?  I know mine doesn't match (missing the last 3 characters - `r0_`).

**Comment by danmosemsft** [comment]
Right, but Windows would keep using the old one. Anyway, this is off topic I guess 😸

**Comment by lifengl** [comment]
I guess the customer has their product name after the user name portion, but cut that part out from the repro before sending it to us.

**Comment by Clockwork-Muse** [comment]
@lifengl - Possibly, but then they should be using [`Environment.GetFolderPath`]https://docs.microsoft.com/en-us/dotnet/api/system.environment.getfolderpath?view=netcore-3.1) (usually, providing `Environment.SpecialFolder.LocalApplicationData`).  Of particular note, `C:\` isn't guaranteed to be the drive the user directory is (solely) located on.  Also, using it (and creating the full path via [`Path.Join`](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.join?view=netcore-3.1)) would have prevented the issue shown here....

**Comment by lifengl** [comment]
Well... i guess their one line console app product is too complicated to do all things correctly ;)

Frankly, those discussions are off the topic. If it is used to triage, well..  let say the invalid path might come from a mistake when the end user sets environment variable '$Home', if their product comes from unix. Or it can be a typo, in an output path inside a msbuild file, or maybe it is coming from a dialog, where the end user input the path to save their data. So it can come from all places.

**Comment by Bosch-Eli-Black** [comment]
I ran into this the other day: A game that I was running managed to create a directory with a pure-whitespace name. When I tried to delete it via Windows Explorer, Explorer just threw up an error. Must be very confusing for normal Windows users, haha.
